src/simsopt/_core/optimizable.py

```diff
@@ -928,30 +1012,27 @@ def full_x(self) -> RealArray:
         with the current Optimizable object and those of its ancestors
         """
         return np.concatenate([opt._dofs.full_x for
-                               opt in (self.ancestors + [self])])
+                               opt in self._unique_dof_opts])
 
     @property
     def local_x(self) -> RealArray:
         """
         Numeric values of the free DOFs associated with this
         Optimizable object
         """
-        return self._dofs.x
+        return self._dofs.free_x
 
     @local_x.setter
     def local_x(self, x: RealArray) -> None:
         """
         Setter for local dofs.
         """
-        self._set_local_x(x)
-        self._set_new_x()
-
-    def _set_local_x(self, x: RealArray) -> None:
         if self.local_dof_size != len(x):
             raise ValueError
-        self._dofs.x = x
+        self._dofs.free_x = x
         if self.local_dof_setter is not None:
             self.local_dof_setter(self, list(self.local_full_x))
+        self.set_recompute_flag()
```

Comment:
@mbkumar I've pushed some more failing unit tests.  Seems to me that we need to set the recompute flag on `self` but also on the `Optimizables` that share `dofs` with `self`